app/src/main/java/com/example/calculator/MainActivity.kt

```diff
@@ -1,11 +1,144 @@
+                        showText(btClick)
+                    }
+                } catch (e: Exception) {
+                    if (btClick.text.toString() in listOf(getString(R.string.plus), getString(R.string.subtract))) {
+                        if (!TextUtils.isEmpty(operandStr.toString()) && operandStr.toString() in operandListDefault) { return }
+                        showText(btClick)
+                    }
+                }
+            }
+
+            getString(R.string.dot) -> {
+                if(!isNewOperand) {
+                    operandStr.append(binding.btDot.text)
+                    binding.tvResult.text = operandStr.toString()
+                }
+            }
+
+            getString(R.string.clear) -> {
+                operandStr.clear()
+                operandStr.append(if (result % 1 == 0.toDouble()) result.toInt().toString() else String.format("%.2f", result))
+                binding.tvResult.text = operandStr.toString()
+            }
+
+            getString(R.string.modulus) -> {
+                val mudulusNumber:Double
+                try {
+                    if(!TextUtils.isEmpty(binding.tvResult.text)) {
+                        mudulusNumber = binding.tvResult.text.toString().toDouble() / 100
+                        binding.tvResult.text = String.format("%.2f" , mudulusNumber)
+                        operandStr.clear()
+                        operandStr.append(binding.tvResult.text)
+                        return
+                    }
+                }catch (e: Exception) { }
+            }
+
+            getString(R.string.negative) -> {
+                if(!TextUtils.isEmpty(binding.tvResult.text)) {
+                    binding.tvResult.text = (binding.tvResult.text.toString().toDouble() * -1).toString()
+                    operandStr.clear()
+                    operandStr.append(binding.tvResult.text)
+                }
+            }
+
+            getString(R.string.equal) -> {
+                resolveOperand()
+            }
+
+            getString(R.string.delete) -> {
+                isNewOperand = false
+                operandList = ArrayList()
+                operandStr.clear()
+                binding.tvResult.text = ""
+                number1 = 0.0
+                number2 = 0.0
+            }
+        }
+    }
+
+    fun resolveOperand() {
+        number2 = operandStr.toString().toDouble()
+        result = if(TextUtils.isEmpty(operand)) number2 else {
+
+            when(operand) {
+                getString(R.string.plus) -> number1 + number2
+                getString(R.string.subtract) -> number1 - number2
+                getString(R.string.multiple) -> number1 * number2
+                getString(R.string.division) -> number1 / number2
+                else -> 0.0
+            }
+        }
+
+        binding.tvResult.text = if(result % 1 == 0.toDouble()) result.toInt().toString() else String.format("%.2f" , result)
```

Comment:
```suggestion
        binding.tvResult.text = if(result % 1 == 0.toDouble()) result.toInt().toString() else getString(R.string.text_result_format , result)
```
trong đó R.string.text_result_format a định nghĩa trong string.xml